src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +33,26 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                FakeManager manager;
+                try
+                {
+                    manager = Fake.GetFakeManager(this.ExpectedValue);
+                }
+                catch (ArgumentException)
+                {
+                    manager = null;
+                }

```

Comment:
For now, the try/catch will do, but eventually we'll probably want to add a `Fake.IsFake` method and/or a `Fake.TryGetFakeManager` method. There's no point in throwing and catching an exception internally when it can easily be avoided.